app/src/main/java/com/hedvig/app/service/DynamicLinkHandler.kt

```diff
@@ -0,0 +1,87 @@
+package com.hedvig.app.service
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.core.content.ContextCompat.startActivity
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import com.hedvig.app.authenticate.LoginStatus
+import com.hedvig.app.feature.loggedin.ui.LoggedInActivity
+import com.hedvig.app.feature.loggedin.ui.LoggedInTabs
+import com.hedvig.app.feature.marketing.ui.MarketingActivity
+import com.hedvig.app.feature.referrals.ReferralsReceiverActivity
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.feature.settings.MarketManager
+
+class DynamicLinkHandler(
+    val context: Context,
+    val marketManager: MarketManager,
+    val startDefaultActivity: (LoginStatus?) -> Unit
+) {
+
+    enum class DynamicLink(val path: String) {
+        REFERRALS("referrals"),
+        DIRECT_DEBIT("direct-debit"),
+        FOREVER("forever"),
+        INSURANCE("insurance")
+    }
+
+    fun handleIntent(
+        intent: Intent,
+        loginStatus: LoginStatus?
+    ) {
+        FirebaseDynamicLinks.getInstance()
+            .getDynamicLink(intent)
+            .addOnSuccessListener { pendingDynamicLinkData ->
+                // This will actually be null in some cases
+                if (pendingDynamicLinkData != null && pendingDynamicLinkData.link != null) {
+                    val link = pendingDynamicLinkData.link
+                    when (link?.pathSegments?.getOrNull(0)) {
```

Comment:
Perhaps the `DynamicLink`-enum could own this code, perhaps with a `DynamicLink::from`-function on the `companion object`?

Reply:
Yes, fixed